domain/src/test/java/org/oppia/domain/oppialogger/loguploader/LogUploadWorkerTest.kt

```diff
@@ -0,0 +1,245 @@
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class LogUploadWorkerTest {
+
+  @Rule
+  @JvmField
+  val mockitoRule: MockitoRule = MockitoJUnit.rule()
+
+  @Inject
+  lateinit var networkConnectionUtil: NetworkConnectionUtil
+
+  @Inject
+  lateinit var fakeEventLogger: FakeEventLogger
+
+  @Inject
+  lateinit var fakeExceptionLogger: FakeExceptionLogger
+
+  @Inject
+  lateinit var oppiaLogger: OppiaLogger
+
+  @Inject
+  lateinit var analyticsController: AnalyticsController
+
+  @Inject
+  lateinit var exceptionsController: ExceptionsController
+
+  @Inject
+  lateinit var logUploadWorkerFactory: LogUploadWorkerFactory
+
+  @Inject
+  lateinit var dataProviders: DataProviders
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  private lateinit var context: Context
+
+  private val eventLogTopicContext = EventLog.newBuilder()
+    .setActionName(EventLog.EventAction.EVENT_ACTION_UNSPECIFIED)
+    .setContext(
+      EventLog.Context.newBuilder()
+        .setTopicContext(
+          EventLog.TopicContext.newBuilder()
+            .setTopicId(TEST_TOPIC_ID)
+            .build()
+        )
+        .build()
+    )
+    .setPriority(EventLog.Priority.ESSENTIAL)
+    .setTimestamp(TEST_TIMESTAMP)
+    .build()
+
+  private val exception = Exception("TEST")
+
+  @Before
+  fun setUp() {
+    networkConnectionUtil = NetworkConnectionUtil(ApplicationProvider.getApplicationContext())
+    setUpTestApplicationComponent()
+    context = InstrumentationRegistry.getInstrumentation().targetContext
+    val config = Configuration.Builder()
+      .setExecutor(SynchronousExecutor())
+      .setWorkerFactory(logUploadWorkerFactory)
+      .build()
+    WorkManagerTestInitHelper.initializeTestWorkManager(context, config)
+  }
+
+  @Test
+  fun testWorker_logEvent_withoutNetwork_enqueueRequest_verifySuccess() {
+    networkConnectionUtil.setCurrentConnectionStatus(NetworkConnectionUtil.ConnectionStatus.NONE)
+    analyticsController.logTransitionEvent(
+      eventLogTopicContext.timestamp,
+      eventLogTopicContext.actionName,
+      oppiaLogger.createTopicContext(TEST_TOPIC_ID)
+    )
+
+    val workManager = WorkManager.getInstance(ApplicationProvider.getApplicationContext())
+
+    val inputData = Data.Builder().putString(
+      LogUploadWorker.WORKER_CASE_KEY,
+      LogUploadWorker.EVENT_WORKER
+    ).build()
+
+    val request: OneTimeWorkRequest = OneTimeWorkRequestBuilder<LogUploadWorker>()
+      .setInputData(inputData)
+      .build()
+
+    workManager.enqueue(request)
+    testCoroutineDispatchers.runCurrent()
+    val workInfo = workManager.getWorkInfoById(request.id)
+
+    assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.SUCCEEDED)
+    assertThat(fakeEventLogger.getMostRecentEvent()).isEqualTo(eventLogTopicContext)
+  }
+
+  @Test
+  fun testWorker_logException_withoutNetwork_enqueueRequest_verifySuccess() {
```

Comment:
It seems also worthwhile to test that when there is network that it doesn't enqueue the request (since it should log it directly).

Reply:
Actually that `withoutNetwork` is basically for putting the exceptions inside the store and the work request will perform its function in presence of network only (in background, when the app instance is closed).

Reply:
To be clear: I'm actually suggesting that we verify that we _don't_ enqueue the request if we log with internet connectivity on (since it should just log as normal). Am I maybe misunderstanding why we shouldn't do that?

Reply:
Okay, so work manager will work in the background only. It'll get enqueued at its specified time interval only and will fetch logs from the cache store. Now when the app is running and there is network, it'll not store in the cache store and will log directly itself. If there are no cached log reports in the store, it'll just complete the process of the request. Does that help ?

Reply:
Yep, I'm actually asking that you verify everything you just said actually works like we expect it to.

Reply:
Work manager and event logging are completely separate things here. Request enqueue and dequeue will work regardless of how we are logging our events. The work manager's task is to enqueue a request, fetch the logs from the store and upload them on Firebase. Even if there is network connection, work request will get enqueued to fetch data from the cache store. Logging of an event/exception is a part of the Analytics/Exception controller and not the work manager.

Reply:
Keeping this open, for further communication in future.